src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
```

Comment:
According to the style guide, constant names should names uppercase. Check and rename in all places.
```suggestion
    private static final int DEF_CAPACITY = 10;
    private static final Object[] EMPTY_ARRAY = {};
```